### PR TITLE
Represent `OngoingMerge` in `MergingRunState`

### PR DESCRIPTION
**Pull request by jorisdral**
# Description

For now, any pattern match on an `OngoingMerge` will lead to an error. This is safe, since we always step merges to completion immediately.

# Checklist

- [x] Read our contribution guidelines at [CONTRIBUTING.md](https://github.com/IntersectMBO/lsm-tree/blob/main/CONTRIBUTING.md), and make sure that this PR complies with the guidelines.

### CODE REVIEW

File: src-extras/Database/LSMTree/Extras/NoThunks.hs [hunk 1]
```diff
@@ -506,7 +506,7 @@ instance NoThunks a => NoThunks (MutVar s a) where
   showTypeOf (_ :: Proxy (MutVar s a)) = "MutVar"
   wNoThunks ctx var = do
       x <- unsafeSTToIO $ readMutVar var
-      noThunks ctx (AllowThunk x) -- TODO: atomicModifyMutVar' is not strict enough
+      noThunks ctx x
```
Comment (dcoutts) [inline src-extras/Database/LSMTree/Extras/NoThunks.hs:509]: So what happened with the problem of `atomicModifyMutVar'`?

**Review by dcoutts — APPROVED** [review]
LGTM!

File: src-extras/Database/LSMTree/Extras/NoThunks.hs (hunk 1 above)
Comment (jorisdral) [reply src-extras/Database/LSMTree/Extras/NoThunks.hs:509]: We used to use it for `RefCounter`, but now we use a `PrimVar` instead, so we're not checking thunks in `MutVar`s anymore

Comment (jorisdral) [reply src-extras/Database/LSMTree/Extras/NoThunks.hs:510]: The problem is still there -- but not allowing a thunk is the proper instance anyway

**Pull request merged**